orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/get_backup_data_sizes.sh

```diff
@@ -0,0 +1,148 @@
+  doLog "Getting the data sizes for the local databases."
+
+  DATABASES='"hive" "ranger" "profiler_agent" "profiler_metric"'
+  RESULT=""
+
+  CUR_DB_SIZE=0
+  TRIMMED_DB=""
+  for DB in $DATABASES; do
+    TRIMMED_DB=$(echo "$DB" | tr -d '"')
+    if doesDatabaseExist "$TRIMMED_DB" ; then
+      CUR_DB_SIZE=$(runPSQLCommand "SELECT pg_database_size('${TRIMMED_DB}');")
+      RESULT="${RESULT}${DB}:${CUR_DB_SIZE},"
+      doLog "Size of database ${DB} is ${CUR_DB_SIZE} bytes."
+    fi
+  done
+
+  doLog "Finished getting the data sizes for the local databases."
+  echo "{${RESULT%?}}"
+}
+
+kinit_as() {
+  doLog "Attempting kinit as $1 using Keytab: $2"
+  kinit -kt "$2" "$1/$(hostname -f)"
+  if [[ $? -ne 0 ]]; then
+    errorExit "Couldn't kinit as $1."
+  fi
+}
+
+getDataSizesForHBase() {
+    doLog "Getting data sizes for HBase tables."
+
+    HBASE_KEYTAB=$(find /run/cloudera-scm-agent/process/ \
+      -name "*.keytab" -a \( -path "*hbase-REGIONSERVER*" -o -path "*hbase-MASTER*" \) \
+      -a -type f | head -n 1)
+    kinit_as "hbase" "$HBASE_KEYTAB"
+
+    # Get sizes of each table along with the name of the table. (Sizes are duplicated which is why we skip the second field)
+    SIZES=($(hdfs --loglevel ERROR  dfs -du -x /hbase/data/default/ 2> >(doLog) | awk '{print $1" "$3}'))
+
+    # Construct final output from response.
+    RESULT=""
+    TABLE_NAME=""
+    CUR_SIZE=""
+    IDX=0
+    {% raw %}
+    while [[ ${IDX} -lt ${#SIZES[@]} ]]; do
+    {% endraw %}
+      CUR_SIZE="${SIZES[$IDX]}"
+      TABLE_NAME="${SIZES[$((IDX+1))]##*/}" # Remove preceding directories in path to table.
+      TABLE_NAME=$(echo ${TABLE_NAME} | awk '{print tolower($0)}')
+      doLog "Size of HBase table \"${TABLE_NAME}\" is ${CUR_SIZE} bytes."
+
+      RESULT="${RESULT}\"${TABLE_NAME}\":${CUR_SIZE},"
+      IDX=$((IDX+2))
+    done
+
+    doLog "Finished getting data sizes for HBase tables."
+    echo "{${RESULT%?}}"
+}
+
+getDataSizesForSolr() {
+  doLog "Getting data sizes for Solr collections."
+
+  SOLR_KEYTAB=$(find /run/cloudera-scm-agent/process/ -name "*.keytab" -path "*solr-SOLR_SERVER*" | head -n 1)
+  kinit_as "solr" "$SOLR_KEYTAB"
+
+  # Obtain sizes for all Solr shards. Below query simply obtains the sizes of each shard for each collection, ensuring to ignore replicas by taking
+  # the maximum size replica, and then summing the sizes of the shards for each collection.
+  SIZES=$(curl -k -s --negotiate -u : "https://$(hostname -f):8985/solr/admin/cores?action=STATUS&wt=json&indent=true" | \
+    jq -c 'reduce .status[] as $x (null; .[$x.cloud.collection][$x.cloud.shard] = ([.[$x.cloud.collection][$x.cloud.shard], $x.index.sizeInBytes] | max))
+    | reduce to_entries[] as $x (null; .[$x.key] += ([$x.value[]] | add))')
+  doLog "Solr collection sizes result: ${SIZES}"
+
+  doLog "Finished getting data sizes for Solr collections."
+  echo "$SIZES"
+}
+
+outputDataSizes() {
+  printf '{"database":%s,"hbase":%s,"solr":%s}\n' "$1" "$2" "$3"
```

Comment:
Can we write this to a separate file, to make it easier to find and parse?
Also I don't know if it's possible, but it would be nice for that file to have a timestamp as a part of the unique name.
that way we have this backup of historical data. We might have to wildcard the permissions however.

Reply:
A few notes on that which aren't meant to be a rejection but just part of a discussion on this since I am not against your ideas:

- Creating multiple files while easily done and good for what you are proposing becomes problematic due to the need to have a certain limit to the amount of space populated by the files, i.e. imagine some customer has done like a few thousand backups and now has a few thousand files containing this info. While the files would not be large it is still up to us to set and enforce limitations to the files we create on the customer's nodes. This is handled by CM via log rotation/cleanup but that seems like a complex approach to this when it's not super necessary for our design.
- I don't mind writing this info into a file, the reason why I chose to write it to stdout was for the CB-side of things, so that the output could be obtained directly from the Salt call to the script. Otherwise I would need to first call the script via Salt, and then make another call to Salt to read the results from the file. I am not against this, in fact it was one of my potential implementations, but I think it should be known since it is a little extra work/communication with the DL.

Reply:
Makes sense. Part of the reason to spit it to a different file would be that the .log would still get replaced with every time. I don't think an extra 1 line file for each backup is going to be the straw that breaks the camel's back. 

For getting the info back to CB, I like your idea here so keeping it on stdout is fine.